File: 434-number-of-segments-in-a-string/number-of-segments-in-a-string.py
class Solution:
    def countSegments(self, s: str) -> int:
        i, c, n = 0, 0, len(s)
        while i < n:
            while i < n and s[i] == " ":
                i += 1
            count = False
            while i < n and s[i] != " ":
                i += 1
                count = True
            if count:
                c += 1
            i += 1
        return c

File: 434-number-of-segments-in-a-string/number-of-segments-in-a-string.cpp
class Solution {
public:
    int countSegments(string s) {
        int n = s.length();
        int count = 0, i = 0;
        while (i < n) {
            while (i < n && s[i] == ' ')
                i += 1;
            if (i >= s.size())
                break;
            while (i < n && s[i] != ' ') 
                i += 1;
                
            count += 1;
        }
        return count;
    }
};